File: 总题库/55.JumpGame.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if (nums.size() == 0) return false;
        
        int numsLen = nums.size();
        int border = nums[0];
        int index = 0;
        while (index < nums.size() && index <= border) {
            if (nums[index] + index > border) {
                border = nums[index] + index;
                if (border >= nums.size() - 1) {
                    return true;
                }
            }
            ++index;
        }
        return false;
    }
};

File: 总题库/55.JumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 0:
            return False
        border = nums[0]
        index = 0
        while index < len(nums) and index <= border:
            border = max(border, index + nums[index])
            if border >= len(nums) - 1:
                return True
            index += 1
            
        return False